### PR TITLE
feature/abstract snapshotting setup

### PR DESCRIPTION
**Pull request by rhyslbw**
Adds snapshotting setup to Space.eventSourcing module, defaults to `enabled` 

Checklist:
- [x] Agree on code style :-) 
- [x] Confirm snapshot default of 10 is good
- [x] Remove processing timeout configuration
- [x] Remove commented out lines in test
- [x] Decide if we're preparing a new major release, or a point release.
- [x] Handle backwards compatibility, in particular the commits collection name.

### CODE REVIEW

File: source/server.coffee [hunk 1]
```diff
@@ -1,41 +1,106 @@
...
 
   @publish this, 'Space.eventSourcing'
 
+  @commitsCollection: null
+  @snapshotsCollection: null
+
+  Configuration: Space.getenv.multi({
+    eventSourcing: {
+      log: {
+        enabled: ['SPACE_ES_LOG_ENABLED', true, 'bool']
```
Comment (DominikGuzei) [inline source/server.coffee]: sure about logging by default?

File: tests/infrastructure/messaging.tests.coffee [hunk 2]
```diff
@@ -2,14 +2,14 @@ describe 'Space.eventSourcing - messaging', ->
 
   customer = id: 'customer_123', name: 'Dominik'
   registration = id: 'registration_123'
-  sharedCommitCollection = new Mongo.Collection('messaging_shared_commits')
+#  sharedCommitCollection = new Mongo.Collection('messaging_shared_commits')
```
Comment (DominikGuzei) [inline tests/infrastructure/messaging.tests.coffee]: can be removed

File: tests/infrastructure/messaging.tests.coffee [hunk 3]
```diff
@@ -2,14 +2,14 @@ describe 'Space.eventSourcing - messaging', ->
...
 
   beforeEach ->
     @fakeDates = sinon.useFakeTimers('Date')
     @app = new CustomerApp()
     @app.configure {
       appId: 'CustomerApp'
-      eventSourcing: { commitsCollection: sharedCommitCollection }
+#      eventSourcing: { commitsCollection: sharedCommitCollection }
```
Comment (DominikGuzei) [inline tests/infrastructure/messaging.tests.coffee]: remove

File: tests/test_application.coffee [hunk 4]
```diff
@@ -6,16 +6,24 @@ via the new configuration api introduced in space:base.
...
   Dependencies: {
     commandBus: 'Space.messaging.CommandBus'
     eventBus: 'Space.messaging.EventBus'
-    Mongo: 'Mongo'
+    mongo: 'Mongo'
   }
 
   Configuration: {
-    useSnapshotting: true
+    testMode: true
```
Comment (DominikGuzei) [inline tests/test_application.coffee]: are we using `testMode` anywhere yet?

File: source/server.coffee [hunk 5]
```diff
@@ -1,41 +1,106 @@
...
+    @injector.get('Space.eventSourcing.Repository').useSnapshotter snapshotter
+
+
+  _collectionOptions: ->
+    if @_externalMongo()
+      if @_externalMongoNeedsOplog()
+        driverOptions = { oplogUrl:  @Configuration.eventSourcing.commits.mongoOplogUrl }
+      else
+        driverOptions = {}
+      return { _driver: new @mongoInternals.RemoteCollectionDriver  @Configuration.eventSourcing.commits.mongoUrl, driverOptions }
```
Comment (DominikGuzei) [inline source/server.coffee]: What about:
```
{ mongoUrl } = @Configuration.eventSourcing.commits
return _driver: new @mongoInternals.RemoteCollectionDriver(mongoUrl, driverOptions)

**Comment by DominikGuzei** [comment]
I think the default value of 10 for snapshotting is fine :wink:

**Comment by DominikGuzei** [comment]
let's keep it at a point release, since nobody is using that in production (strong guess, but still) :wink:

File: source/server.coffee (hunk 1 above)
Comment (rhyslbw) [inline source/server.coffee]: Ah no, that's a mistake. Good catch

File: tests/test_application.coffee (hunk 4 above)
Comment (rhyslbw) [inline tests/test_application.coffee]: Not yet so will remove. I'll document the Configuration API [here](https://meteor-space.readme.io/v1.0/docs/configuration) which can be our master reference, including a specification of what has been implemented

File: tests/test_application.coffee [hunk 6]
```diff
@@ -6,16 +6,24 @@ via the new configuration api introduced in space:base.
...
+    eventSourcing: {
+      snapshotting: {
+        frequency: 2
+        collectionName: 'my_collection'
```
Comment (rhyslbw) [inline tests/test_application.coffee]: I think we should remove this and let the default apply, then rig it in a specific test to check that functionality.

Comment (DominikGuzei) [inline tests/test_application.coffee]: agreed

File: tests/test_application.coffee (hunk 4 above)
Comment (DominikGuzei) [inline tests/test_application.coffee]: ok nice! :+1:

**Comment by DominikGuzei** [comment]
Ok, so the travis CI tests would actually work – but you have to use `https` urls for the package deps in `git-packages.json`. And i think there are some things you haven't pushed yet to `space:base`?

**Comment by rhyslbw** [comment]
I agree we'll need to publish the next version before we have enough for a major release, but am a little concerned about this as it's a breaking change for anyone using the package rather than an obscure API. I'll see if we can rig compatibility that can then be removed at the next major version.

**Comment by DominikGuzei** [comment]
Ok, although i really don't think anyone is using this in production yet (the only place where this would really be a problem). But you're right – it's the "correct" way to got :wink:

**Pull request merged**